atomic_reactor/plugins/post_koji_import.py

```diff
@@ -613,20 +564,126 @@ def _filesystem_koji_task_id(self) -> Optional[int]:
             self.log.error("invalid task ID %r", fs_task_id, exc_info=1)
             return None
 
-    def get_output(self, buildroot_id):
-        """
-        Build the output entry of the metadata.
+    def _collect_source_tarballs_outputs(
+            self, buildroot_id: str, outputs_metadata: List[Dict[str, Any]]
+    ) -> None:
+        wf_data = self.workflow.data
+        # a list of metadata describing the remote sources.
+        plugin_results: List[Dict[str, Any]]
+        plugin_results = wf_data.prebuild_results.get(PLUGIN_RESOLVE_REMOTE_SOURCE) or []
+        for remote_source in plugin_results:
+            remote_source_tarball = remote_source['remote_source_tarball']
+            local_filename = remote_source_tarball['path']
+            dest_filename = remote_source_tarball['filename']
+
+            metadata = get_output_metadata(local_filename, dest_filename)
+            metadata['buildroot_id'] = buildroot_id
+            add_type_info(metadata, KOJI_BTYPE_REMOTE_SOURCES)
+            outputs_metadata.append(metadata)
+
+            wf_data.koji_upload_files.append({
+                'local_filename': local_filename,
+                'dest_filename': dest_filename,
+            })
+
+    def _collect_remote_sources_json_outputs(
+            self, buildroot_id: str, outputs_metadata: List[Dict[str, Any]]
+    ) -> None:
```

Comment:
I'd prefer just one method for collecting remote source metadata instead of having one for tarballs and one for JSON output